### PR TITLE
config: introduce a config client to support load configs from PD online

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve? 
Part I to implement this proposal: https://github.com/pingcap/tidb/pull/13660.
Wait for https://github.com/pingcap/pd/pull/2041.
In this PR, we introduce a config client(`ConfHandler`) to support load configs from PD online.

### What is changed and how it works?
1. Move the method `parsePath` from `kv` package to `config` package to avoid recycle import.
2. Add a new method `MarshalText` to `nullableBool` to make it can be encoded and decoded correctly.
3. Introduce `ConfHandler` to support to load configs from PD online.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: config/config_handler.go [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+	}
+
+	// register to PD and get the new default config.
+	// see https://github.com/pingcap/tidb/pull/13660 for more details.
+	// suppose port and security config items cannot be change online.
+	status, version, conf, err := pdCli.Create(context.Background(), new(configpb.Version), tidbComponentName, id, confContent)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if status.Code != configpb.StatusCode_OK {
```
Comment (rleungx) [inline config/config_handler.go]: `StatusCode_WRONG_VERSION` will return the latest version and config. It should not be returned here?

File: config/config_handler.go [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+			// fetch new config from PD
+			status, version, newConfContent, err := ch.pdConfCli.Get(context.Background(), ch.version, tidbComponentName, ch.id)
+			if err != nil {
+				logutil.Logger(context.Background()).Error("[PDConfHandler] fetch new config error", zap.Error(err))
+				continue
+			}
+			if status.Code == configpb.StatusCode_NOT_CHANGE {
+				continue
+			}
+			if status.Code != configpb.StatusCode_OK {
```
Comment (rleungx) [inline config/config_handler.go]: ditto

File: config/config_handler.go [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+		}
+		ch.wg.Done()
+	}()
+	for {
+		select {
+		case <-time.After(ch.interval):
+			// fetch new config from PD
+			status, version, newConfContent, err := ch.pdConfCli.Get(context.Background(), ch.version, tidbComponentName, ch.id)
+			if err != nil {
+				logutil.Logger(context.Background()).Error("[PDConfHandler] fetch new config error", zap.Error(err))
```
Comment (XuHuaiyu) [inline config/config_handler.go]: I think we do not need `[PDConfHandler]`

`Error("pdConfHandler fetches new config fail", zap.Error(err))` is enough

File: config/config_handler.go [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+			if err != nil {
+				logutil.Logger(context.Background()).Error("[PDConfHandler] decode config error", zap.Error(err))
+				continue
+			}
+			if err := newConf.Valid(); err != nil {
+				logutil.Logger(context.Background()).Error("[PDConfHandler] invalid config", zap.Error(err))
+				continue
+			}
+
+			if ch.reloadFunc != nil {
```
Comment (XuHuaiyu) [inline config/config_handler.go]: `reloadFunc` will not be nil, we can remove this check.

File: config/config_handler.go [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+	ch.wg.Wait()
+	ch.pdConfCli.Close()
+}
+
+func (ch *pdConfHandler) GetConfig() *Config {
+	return ch.curConf.Load().(*Config)
+}
+
+func (ch *pdConfHandler) run() {
+	defer func() {
```
[the remark this answers is not in the document]
Comment (qw4990) [reply config/config_handler.go]: After testing, I found that using `util` will cause cycle import, so I keep the native defer function now.

**Comment by qw4990** [comment]
@XuHuaiyu @rleungx All comments have been addressed, PTAL~

File: config/config.go [hunk 6]
```diff
@@ -864,3 +876,28 @@ const (
...
+// ParsePath parses this path.
+func ParsePath(path string) (etcdAddrs []string, disableGC bool, err error) {
+	var u *url.URL
+	u, err = url.Parse(path)
+	if err != nil {
+		err = errors.Trace(err)
+		return
+	}
+	if strings.ToLower(u.Scheme) != "tikv" {
+		err = errors.Errorf("Uri scheme expected[tikv] but found [%s]", u.Scheme)
```
Comment (XuHuaiyu) [inline config/config.go]: add a blank between `expected` and `[tikv]`

**Comment by rleungx** [comment]
CI failed.

File: config/config.go (hunk 6 above)
Comment (qw4990) [reply config/config.go]: updated.

**Comment by qw4990** [comment]
@rleungx All comments have been addressed, PTAL. And please help me to approve this PR if there is no problem.

**Pull request merged**